### PR TITLE
Add audio previews to map pages

### PR DESCRIPTION
**Pull request by Pixelguymm**

### CODE REVIEW

File: src/jsMain/kotlin/io/beatmaps/util/audio.kt [hunk 1]
```diff
@@ -8,10 +8,10 @@ fun useAudio() = useRef(
     Audio().also {
         it.volume = 0.4
     }
-).also { audio ->
+).also {
     useEffectOnce {
         cleanup {
-            audio.current?.pause()
+            it.current?.pause()
```
Comment (Top-Cat) [inline src/jsMain/kotlin/io/beatmaps/util/audio.kt:14]: Any reason?

File: src/jsMain/kotlin/io/beatmaps/shared/audioPreview.kt [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                    it.pause()
+                    it.currentTime = 0.0
+                }
+            }
+        }
+    }
+
+    override fun RBuilder.render() {
+        div("audio-progress") {
+            attrs.jsStyle["height"] = props.size + "px"
```
Comment (Top-Cat) [inline src/jsMain/kotlin/io/beatmaps/shared/audioPreview.kt:96]: Clearer as a block imo
```
attrs.jsStyle {
    height = props.size + "px"
    width = props.size + "px"
}
```

**Comment by Top-Cat** [comment]
Not sure this is quite right?
Screenshot 2023-11-11 at 13 42 32

File: src/jsMain/kotlin/io/beatmaps/util/audio.kt (hunk 1 above)
Comment (Pixelguymm) [reply src/jsMain/kotlin/io/beatmaps/util/audio.kt:14]: Moved it to a file where it shadowed another variable. Then moved it back and forgot to change it. I'll revert that.

**Comment by Pixelguymm** [comment]
> Not sure this is quite right?

I couldn't get uploads to work locally, despite having all the folders. That's the one part I couldn't test -_-.

**Pull request merged**